src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -364,6 +368,8 @@ void BaseCouplingScheme::secondExchange()
           PRECICE_DEBUG("Setting require create checkpoint");
           requireAction(CouplingScheme::Action::WriteCheckpoint);
         }
+        _totalTimeDrift += _currentWindowTimeDrift;
+        PRECICE_CHECK(abs(_totalTimeDrift) < math::NUMERICAL_ZERO_DIFFERENCE, "preCICE has detected a difference between its internal time and the time of this participant. This can happen, if you are using very many substeps per time window over multiple time windows. Please refer to https://github.com/precice/precice/issues/1866 for strategies to avoid this problem.");
```

Comment:
Yes. We can remove the `abs` here and use the `unsigned double _totalTimeDrift`.

I would say we should error, because everything else will stay unnoticed and if the time drift becomes too large we should exit. Otherwise we might end up with situations like the one described in https://github.com/precice/precice/issues/1866.

On documentation: We definitely need to explain this better, but I first would like to conclude on this issue and on #1882 before investing time into documentation. There is also a potential refactoring for `_computedTimeWindowPart` on the way, because the whole situation here is very fuzzy as well.

Reply:
I added some corresponding documentation here: https://github.com/precice/precice.github.io/pull/257/files#diff-d807d6cfae4162be6a479b8c0280b99775fafb443f95d6fee3fb19619a6391dc

Currently, this is a draft, but we can first merge this PR here and then take care of the documentation later. I definitely have a more general review of all documentation regarding subcycling and time stepping in mind.